be/src/vec/exprs/vexpr_context.cpp

```diff
@@ -105,6 +105,23 @@ doris::Status VExprContext::clone(RuntimeState* state, VExprContext** new_ctx) {
     return _root->open(state, *new_ctx, FunctionContext::THREAD_LOCAL);
 }
 
+doris::Status VExprContext::clone(RuntimeState* state, std::shared_ptr<VExprContext>& new_ctx) {
```

Comment:
maybe we could define ExprSPtr and changing both expr and exprcontext as shared ptr, then we do not need add new method, just modify the original method. 
because I find there are many duplicate code...